spec/features/user_edits_book_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+feature 'User edits an existing book' do
+  scenario 'updates title field successfully' do
+    @book = create(:book)
+    visit root_path
+
+    edit_book(@book, new_title: 'The Golden Compass',
+                     new_author: @book.author)
+
+    expect(page.current_path).to eq books_path
```

Comment:
I would combine these two tests into one test that sets a new author and a new title at once. Having two separate tests doesn't give you much, but will slow things down and make it so you have to update two tests if something in the app requires the edit page to change. I would also test that the page has the new title and author on the page to make sure it was updated. Right now you could skip updating in the controller and the test would still pass